src/test/java/uk/gov/pay/adminusers/resources/ServiceResourceCreateTest.java

```diff
@@ -43,26 +39,19 @@ public void shouldSuccess_whenCreateAServiceWithValidGatewayAccounts() throws Ex
                 .put("gateway_account_ids", new String[]{"1", "2"})
                 .build();
 
-        ValidatableResponse validatableResponse = givenSetup()
+        givenSetup()
                 .when()
                 .accept(JSON)
                 .body(mapper.writeValueAsString(payload))
                 .post("/v1/api/services")
                 .then()
-                .statusCode(201);
-
-        validatableResponse
+                .statusCode(201)
                 .body("name", is("System Generated"))
-                .body("external_id", notNullValue());
-
-        String serviceExternalId = validatableResponse.extract().jsonPath().getString("external_id");
-
-        List<Map<String, Object>> gatewayAccountsForService = databaseHelper.findGatewayAccountsByService(serviceExternalId);
```

Comment:
The response check now checks of the gateway_account_ids returned as part of the response. The fact that it works proves the ID's are saved in database. Before we had to get to the database to check because we never returned the saved object.